### ISSUE TITLE
Dicom File with UN length will not open

### ISSUE THREAD

**Issue by keithwoodward** [fo-dicom/fo-dicom#1057]
### latest fo dicom version 4.0.5

[IM-0007-0276-586.dcm.zip](https://github.com/fo-dicom/fo-dicom/files/4826347/IM-0007-0276-586.dcm.zip)

**Comment by gofal** [comment]
the file is definitively broken. DicomTag 0008,1250, Related Series Sequence, is invalid.
An other application i tried just skipped this tag. 

fo-dicom has the philosophy to be as strict as possible when creating data, and as tollerant as possible when receiving data. There are quite some fixes where fo-dicom corrects an error if it is obvious how to fix it. But this tag is corruct and fo-dicom cannot correct it.
I am not sure if it is fine to silently remove a DicomTag and then continue as if everything was fine. We are talking about medical data. if the file is broken, then you should reject and blame the source that the file is corrupt. They should have to fix the problem.

**Comment by keithwoodward** [comment]
Understood.  Somewhat disappointing that fo dicom won’t open files that many other viewers can open.

Seems like it should have an option to skip bad tags when opening a file, I guess SkipInvalidTags = True type option, so the user can use fo-dicom on as many images as possible if they need to?

Unfortunately blaming the source isn’t always possible.


Thank you for the response.  I think fo dicom is a great package!


Keith

**Comment by gofal** [comment]
yes, thats an issue that is raised from time to time. An idea has been that a DicomParseException should be thrown when reading such a dataset, and this exception has a property of type DicomDataset that contains all successfully read DicomTags. So you can continue to work but should be aware that this dataset is not complete.
or if there is some invalid DicomTag, then the parser should insert some pseudo-tag into the dataset and the exception is then not thrown when parsing the dataset, but when first accessing this DicomTag. But that would mean, you can read the data, but cannot store and send it again.
Discussions then always stopped somehow and we never found a good solution.

**Comment by keithwoodward** [comment]
Agree that sounds like a good plan.

If its a lack of funding, I can maybe help with that?  I will now have to buy a commercial package to open the files and clean them before fo-dicom tries to open them.

Thanks again,
Keith

**Comment by amoerie** [comment]
Not sure if you are a developer Keith, but I have had good results with maintaining a private clone of fo-dicom that contains the features we need at my company. It is quite easy to have a git repository with multiple remotes, one for the official repo and one to your private repo. This facilitates keeping your private copy in sync. Of course that would still mean you'd have to develop this yourself.

It's not necessarily a lack of funding. We're simply a small group of volunteers who, from time to time, contribute to this project. There isn't any system in place for "sponsored development" or "paid priority features" because, well, I think you might be the first person asking for these things.

**Comment by mrbean-bremen** [comment]
I had a look at the dataset in question, and the reason why other frameworks handle this differently is the handling of VR `UN` for explicit transfer syntax. `fo-dicom` always ignores the `UN` for known tags and uses the correct VR for that tag, while other frameworks  (I know at least of dcmtk and pydicom) use a setting for this, that defaults to handle the tag as is (e.g. as `UN`). We could probably add such a setting in `fo-dicom`, which would it make possible to handle this kind of problem. 
This is of course a rare case, and I'm not sure if it's worth to consider (and this issue is so old that it surely won't matter for this concrete issue). I also don't know where this kind of option would be configured in fo-dicom. 

Anyway, I wanted to mention it in case it would make sense. The actual line where this is handled is this line in `DicomReader.cs`:
```cs
    if (_vr == DicomVR.Implicit || (_vr == DicomVR.UN && _isExplicitVR))
```
which could easily be adapted by adding a check to the second part. As this happens rarely, this would also not impact the performance.

**Comment by gofal** [comment]
This will be fixed with PR #1911
